apis/r/R/TileDBCreateOptions.R

```diff
@@ -0,0 +1,257 @@
+# Example config (written JSON-style for compact reference -- see unit-test files for precise
+# syntax):
+#
+# platform_config = {
+#     "tiledb": {
+#         "create": {
+#             "cell_order": "ROW_MAJOR",
+#             "tile_order": "COL_MAJOR",
+#             "capacity": 8888,
+#             "dataframe_dim_zstd_level": 1,
+#             "sparse_nd_array_dim_zstd_level": 2,
+#             "offsets_filters": ["RLE", "NONE"],
+#             "dims": {
+#                 "soma_dim_0": {"tile": 6},
+#                 # Empty filters for soma_dim_1 overrides the default
+#                 # dimension zstd level defined below.
+#                 "soma_dim_1": {"filters": []},
+#             },
+#             "attrs": {"soma_data": {"filters": ["RLE"]}},
+#         }
+#     }
+# }
+
+# Non-filter-related schema parameters
+DEFAULT_TILE_ORDER  <- function() { "ROW_MAJOR" }
+DEFAULT_CELL_ORDER  <- function() { "ROW_MAJOR" }
+DEFAULT_TILE_EXTENT <- function() { 2048 }
+DEFAULT_CAPACITY    <- function() { 100000 }
```

Comment:
Functions returning consts works, but may be heavier?  A not-uncommon paradigm is just values in an evnvironment that is global to the package, unexported otherwise.  But this works too and is simple to grok and change.

Reply:
@eddelbuettel i'm here for feedback :)

i googled 'how to make constants in R' or somesuch & found this approach. happy to go with whatever's most idiomatic.

Reply:
Switching to DM :grinning: